### PR TITLE
DR-1509 Distinct the files before attempting to update the dependencies FS collection

### PR DESCRIPTION
**Pull request by nmalfroy**
Note: this change is minimal.  There are likely other places that we could use the distincted list but want to get this initial version in first

### CODE REVIEW

**Review by fboulnois — COMMENTED** [review]
This change makes sense to solve Raaid's issue, but what is the larger implication of this change to our `refIds`?

**Comment by nmalfroy** [comment]
So, I left the refIds alone.  Currently, this only affect when we write to the <datasetid>-dependencies firestore collection and that only contains snapshot+fileId. The document has a ref count but I don't know that we use that for anything and even what that represents (e.g. it looks like the only valid values are 0 or 1 so it makes me feel like this change is the correct one)

**Review by snf2ye — APPROVED** [review]

**Comment by ddietterich** [comment]
It is not the case that the only valid values are 0 and 1. With 1536 dups, we will have several batches that will count the same file_id. That will result in a refCount of > 1.

We never call the remove method (as we used to in earlier version of this code), so I am pretty sure the only thing that matters is that the dependency row exists or does not exist.

The next improvement would be to remove the refCount altogether. Then the dependency batch would only need to set when the file_id did not already have a dependency entry in the collection.

File: src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFireStoreDataStep.java [hunk 1]
```diff
@@ -72,6 +76,11 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
                             mapTable.getFromTable().getName(),
                             mapTable.getFromTable().getId().toString(),
                             mapColumn.getFromColumn());
+                        List<String> uniqueRefIds = refIds.stream().distinct().collect(Collectors.toList());
```
Comment (ddietterich) [inline src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFireStoreDataStep.java:79]: Hmm... I would have put this inside of the dependencyDao and not here. It'll do for now.

**Review by ddietterich — APPROVED** [review]
ok by me

File: src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFireStoreDataStep.java (hunk 1 above)
Comment (nmalfroy) [reply src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFireStoreDataStep.java:79]: Yeah, I was thinking that we may want to reuse the distincted values in `addFilesToSnapshot`

**Comment by nmalfroy** [comment]
Ah ok, I see.  Yeah, maybe we can rework that logic later too

**Comment by nmalfroy** [comment]
Oh I see...should it do a count of file references?  Is it going to be ok for it to not have the refcount be accurate?

**Comment by ddietterich** [comment]
As far as I can tell looking at the code, it should be OK for the count to be inaccurate. The decrement code appears unused and the dependency check code uses document existence - not the refCount.

**Comment by nmalfroy** [comment]
Got it cool.  Will merge once tests pass and bump fakeprod

**Pull request merged**